### PR TITLE
[Imgtool] More C++-ification (imgtool_stream ==> imgtool::stream)

### PR DESCRIPTION
**Pull request by npwoods**
Still a bit of work to do, most notably around adoption of std::unique_ptr<>, which may be a challenge here on account of some assumptions regarding lifetime of streams by modules.

### CODE REVIEW

File: src/tools/imgtool/iflopimg.cpp [hunk 1]
```diff
@@ -45,29 +45,29 @@ imgtoolerr_t imgtool_floppy_error(floperr_t err)
 
 static void imgtool_floppy_closeproc(void *file)
 {
-	stream_close((imgtool_stream *) file);
+	delete (imgtool::stream *) file;
```
Comment (cuavas) [inline src/tools/imgtool/iflopimg.cpp]: Could you please make these use `reinterpret_cast` to make it clear what kind of cast you're attempting.

File: src/tools/imgtool/iflopimg.cpp [hunk 2]
```diff
@@ -137,28 +137,28 @@ static imgtoolerr_t imgtool_floppy_open_internal(imgtool::image *image, imgtool_
...
 {
 	floperr_t ferr;
 	imgtoolerr_t err = IMGTOOLERR_SUCCESS;
 	struct imgtool_floppy_image *fimg;
 	const imgtool_class *imgclass;
 	const struct FloppyFormat *format;
-	imgtoolerr_t (*create)(imgtool::image *, imgtool_stream *, util::option_resolution *);
-	imgtoolerr_t (*open)(imgtool::image *image, imgtool_stream *f);
+	imgtoolerr_t (*create)(imgtool::image *, imgtool::stream *, util::option_resolution *);
+	imgtoolerr_t (*open)(imgtool::image *image, imgtool::stream *f);
```
Comment (cuavas) [inline src/tools/imgtool/iflopimg.cpp]: Is there any reason the parameters can't be references?

File: src/tools/imgtool/imghd.cpp [hunk 3]
```diff
@@ -169,7 +169,7 @@ void imghd_close(struct mess_hard_disk_file *disk)
 		disk->hard_disk = nullptr;
 	}
 	if (disk->stream)
-		stream_close(disk->stream);
+		delete disk->stream;
```
Comment (cuavas) [inline src/tools/imgtool/imghd.cpp]: Wouldn't it be safer to also set `disk->stream` to `nullptr` when deleting?

File: src/tools/imgtool/modules/thomson.cpp [hunk 4]
```diff
@@ -693,7 +693,7 @@ static void thom_get_file(thom_floppy* f, unsigned head,
 		int i;
 		for ( i = 0; i < 8; i++ ) {
 	UINT8* data = thom_get_sector( f, head, track, firstsect + i );
-	stream_write( dst, data, f->sectuse_size );
+	dst->write(data, f->sectuse_size );
```
Comment (cuavas) [inline src/tools/imgtool/modules/thomson.cpp]: You've half-adjusted the spacing around the parentheses in the call - either do it at both ends or don't do it at all.  Same goes for the rest of the changes in this file.

File: src/tools/imgtool/modules/thomson.cpp [hunk 5]
```diff
@@ -1397,28 +1397,28 @@ static imgtoolerr_t thom_basic_read_file(imgtool::partition *part,
 
 	err = thomcrypt_read_file( part, name, fork, org );
 	if ( err ) {
-	stream_close( org );
+	delete org;
```
Comment (cuavas) [inline src/tools/imgtool/modules/thomson.cpp]: No excuse for not fixing bad indentation if you're changing the lines anyway...

File: src/tools/imgtool/iflopimg.cpp (hunk 2 above)
Comment (npwoods) [inline src/tools/imgtool/iflopimg.cpp]: No, there isn't - I agree that it is desirable to do so, my concern is about "doubling down" on the intrusiveness of this particular change.  This is the same reason that I transitioned imgtool::image and imgtool::partition separately.

In addition to better adoption of references, there other things I'd like to do in follow up patches - change to std::function, change char buffers to std::string, make functions return unique_ptr etc.

**Pull request merged**